dkc/core/rest/folder.py

```diff
@@ -43,14 +42,31 @@ class Meta:
             'public',
             'access',
         ]
+        # ModelSerializer cannot auto-generate validators for model-level constraints
+        validators = [
+            serializers.UniqueTogetherValidator(
+                queryset=Folder.objects.all(),
+                fields=['parent', 'name'],
+                message='Folder names must be unique among siblings.',
```

Comment:
This error message is a little technical -- keep in mind, in the SPA I just display these messages as-is to end users. How about this instead?

"A folder with that name already exists here."